L1Trigger/CSCTriggerPrimitives/src/CSCAnodeLCTProcessor.cc

```diff
@@ -1276,8 +1273,33 @@ CSCALCTDigi CSCAnodeLCTProcessor::getBestALCT(int bx) const { return bestALCT[bx
 
 CSCALCTDigi CSCAnodeLCTProcessor::getSecondALCT(int bx) const { return secondALCT[bx]; }
 
+/** return vector of CSCShower digi **/
+std::vector<CSCShowerDigi> CSCAnodeLCTProcessor::getAllShower() const {
+  std::vector<CSCShowerDigi> vshowers(anode_showers_, anode_showers_ + CSCConstants::MAX_ALCT_TBINS);
+  return vshowers;
+};
+
 /** Returns shower bits */
-CSCShowerDigi CSCAnodeLCTProcessor::readoutShower() const { return shower_; }
+std::vector<CSCShowerDigi> CSCAnodeLCTProcessor::readoutShower() const {
+  unsigned minbx_readout = CSCConstants::LCT_CENTRAL_BX - l1a_window_width / 2;
```

Comment:
Would a check that `CSCConstants::LCT_CENTRAL_BX ->= l1a_window_width / 2` be worth (in the constructor)?

It is true that if `minbx_readout` overflows, the loop at [L1289](https://github.com/cms-sw/cmssw/pull/38941/files#diff-bbb72f657aa720b2f0097ffdd8af20ba5d3940490879aae6c19c2f18105279c6R1289) does not even start, and therefore it is harmless anyhow

Reply:
moved to `minbx_readout` and `maxbx_readout` to header files and added `assert` in the constructor 712e670